### PR TITLE
Remove some stuttering on returns errors

### PR DESCRIPTION
**Pull request by rhatdan**
Golangs os.* functions return the name of the file/directory they
fail to use.  We should not wrap these errors with the file/directory
to use names, causes stuttering when the user sees the errors, and looks
bad having huge error messages.

Since this is just code cleanup, existing tests should handle the
changes.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/blobcache/blobcache.go [hunk 1]
```diff
@@ -215,7 +215,7 @@ func (s *blobCacheSource) GetManifest(ctx context.Context, instanceDigest *diges
 		}
 		if !os.IsNotExist(err) {
 			s.cacheErrors++
-			return nil, "", errors.Wrapf(err, "error checking for manifest file %q", filename)
+			return nil, "", errors.Wrapf(err, "checking for manifest file")
```
Comment (vrothberg) [inline pkg/blobcache/blobcache.go:218]:
```suggestion
			return nil, "", errors.Wrap(err, "checking for manifest file")
```

File: pkg/blobcache/blobcache.go [hunk 2]
```diff
@@ -245,7 +245,7 @@ func (s *blobCacheSource) GetBlob(ctx context.Context, blobinfo types.BlobInfo,
 				s.mu.Lock()
 				s.cacheErrors++
 				s.mu.Unlock()
-				return nil, -1, errors.Wrapf(err, "error checking for cache file %q", filepath.Join(s.reference.directory, filename))
+				return nil, -1, errors.Wrapf(err, "checking for cache")
```
Comment (vrothberg) [inline pkg/blobcache/blobcache.go:248]:
```suggestion
				return nil, -1, errors.Wrap(err, "checking for cache")
```

File: pkg/overlay/overlay.go [hunk 3]
```diff
@@ -174,15 +174,15 @@ func recreate(contentDir string) error {
 		if os.IsNotExist(err) {
 			return nil
 		}
-		return errors.Wrapf(err, "failed to stat overlay upper %s directory", contentDir)
+		return errors.Wrapf(err, "failed to stat overlay upper directory")
```
Comment (vrothberg) [inline pkg/overlay/overlay.go:177]:
```suggestion
		return errors.Wrap(err, "failed to stat overlay upper directory")
```

File: pkg/overlay/overlay.go [hunk 4]
```diff
@@ -174,15 +174,15 @@ func recreate(contentDir string) error {
...
 	}
 
 	if err := os.RemoveAll(contentDir); err != nil {
-		return errors.Wrapf(err, "failed to cleanup overlay %s directory", contentDir)
+		return err
 	}
 
 	if err := idtools.MkdirAllAs(contentDir, os.FileMode(st.Mode()), int(st.UID()), int(st.GID())); err != nil {
-		return errors.Wrapf(err, "failed to create the overlay %s directory", contentDir)
+		return errors.Wrapf(err, "failed to create overlay directory")
```
Comment (vrothberg) [inline pkg/overlay/overlay.go:185]:
```suggestion
		return errors.Wrap(err, "failed to create overlay directory")
```

File: pkg/overlay/overlay.go [hunk 5]
```diff
@@ -218,7 +218,7 @@ func CleanupContent(containerDir string) (Err error) {
 	}
 
 	if err := os.RemoveAll(contentDir); err != nil && !os.IsNotExist(err) {
-		return errors.Wrapf(err, "failed to cleanup overlay %s directory", contentDir)
+		return errors.Wrapf(err, "failed to cleanup overlay directory")
```
Comment (vrothberg) [inline pkg/overlay/overlay.go:221]:
```suggestion
		return errors.Wrap(err, "failed to cleanup overlay directory")
```

File: pkg/blobcache/blobcache.go [hunk 6]
```diff
@@ -140,7 +140,7 @@ func (r *blobCacheReference) HasBlob(blobinfo types.BlobInfo) (bool, int64, erro
 			return true, fileInfo.Size(), nil
 		}
 		if !os.IsNotExist(err) {
-			return false, -1, errors.Wrapf(err, "error checking size of %q", filename)
+			return false, -1, err
```
Comment (TomSweeneyRedHat) [inline pkg/blobcache/blobcache.go:143]: Might make sense to leave behind the bit about checking size

File: pkg/blobcache/blobcache.go (hunk 1 above)
Comment (rhatdan) [reply pkg/blobcache/blobcache.go:218]: Fixed

File: pkg/blobcache/blobcache.go (hunk 2 above)
Comment (rhatdan) [reply pkg/blobcache/blobcache.go:248]: Fixed

File: pkg/overlay/overlay.go (hunk 3 above)
Comment (rhatdan) [reply pkg/overlay/overlay.go:177]: Fixed

File: pkg/overlay/overlay.go (hunk 4 above)
Comment (rhatdan) [reply pkg/overlay/overlay.go:185]: Fixed

File: pkg/overlay/overlay.go (hunk 5 above)
Comment (rhatdan) [reply pkg/overlay/overlay.go:221]: Fixed

File: pkg/blobcache/blobcache.go (hunk 6 above)
Comment (rhatdan) [reply pkg/blobcache/blobcache.go:143]: Fixed

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL

File: pkg/blobcache/blobcache.go [hunk 7]
```diff
@@ -154,7 +154,7 @@ func (r *blobCacheReference) Directory() string {
 func (r *blobCacheReference) ClearCache() error {
 	f, err := os.Open(r.directory)
 	if err != nil {
-		return errors.Wrapf(err, "error opening directory %q", r.directory)
+		return err
```
Comment (nalind) [inline pkg/blobcache/blobcache.go:157]:
```suggestion
		return errors.WithStack(err)
```

File: pkg/overlay/overlay.go [hunk 8]
```diff
@@ -174,15 +174,15 @@ func recreate(contentDir string) error {
 		if os.IsNotExist(err) {
 			return nil
 		}
-		return errors.Wrapf(err, "failed to stat overlay upper %s directory", contentDir)
+		return errors.Wrap(err, "failed to stat overlay upper directory")
 	}
 
 	if err := os.RemoveAll(contentDir); err != nil {
-		return errors.Wrapf(err, "failed to cleanup overlay %s directory", contentDir)
+		return err
```
Comment (nalind) [inline pkg/overlay/overlay.go:181]:
```suggestion
		return errors.WithStack(err)
```

File: pkg/parse/parse.go [hunk 9]
```diff
@@ -178,11 +178,11 @@ func parseSecurityOpts(securityOpts []string, commonOpts *define.CommonBuildOpti
 			commonOpts.SeccompProfilePath = SeccompOverridePath
 		} else {
 			if !os.IsNotExist(err) {
-				return errors.Wrapf(err, "can't check if %q exists", SeccompOverridePath)
+				return err
```
Comment (nalind) [inline pkg/parse/parse.go:181]:
```suggestion
				return errors.WithStack(err)
```

File: pkg/parse/parse.go [hunk 10]
```diff
@@ -178,11 +178,11 @@ func parseSecurityOpts(securityOpts []string, commonOpts *define.CommonBuildOpti
...
 			}
 			if _, err := os.Stat(SeccompDefaultPath); err != nil {
 				if !os.IsNotExist(err) {
-					return errors.Wrapf(err, "can't check if %q exists", SeccompDefaultPath)
+					return err
```
Comment (nalind) [inline pkg/parse/parse.go:185]:
```suggestion
					return errors.WithStack(err)
```

File: pkg/parse/parse.go [hunk 11]
```diff
@@ -454,7 +454,7 @@ func ValidateVolumeHostDir(hostDir string) error {
 	}
 	if filepath.IsAbs(hostDir) {
 		if _, err := os.Stat(hostDir); err != nil {
-			return errors.Wrapf(err, "error checking path %q", hostDir)
+			return err
```
Comment (nalind) [inline pkg/parse/parse.go:457]:
```suggestion
			return errors.WithStack(err)
```

File: pkg/parse/parse.go [hunk 12]
```diff
@@ -468,7 +468,7 @@ func validateVolumeMountHostDir(hostDir string) error {
 		return errors.Errorf("invalid host path, must be an absolute path %q", hostDir)
 	}
 	if _, err := os.Stat(hostDir); err != nil {
-		return errors.Wrapf(err, "error checking path %q", hostDir)
+		return err
```
Comment (nalind) [inline pkg/parse/parse.go:471]:
```suggestion
		return errors.WithStack(err)
```

**Comment by TomSweeneyRedHat** [comment]
LGTM once @nalind 's concerns are addressed.

**Comment by rhatdan** [comment]
@nalind @vrothberg PTAL

**Comment by rhatdan** [comment]
@nalind Could the WithStack calls potentially cause the hang we are seeing in conformance testing?

**Comment by rhatdan** [comment]
@saschagrunert @vrothberg @giuseppe PTAL

**Review by TomSweeneyRedHat — COMMENTED** [review]
LGTM

**Comment by nalind** [comment]
> @nalind Could the WithStack calls potentially cause the hang we are seeing in conformance testing?

Highly unlikely.  Wrap() and Wrapf() call the same underlying function, so switching from those to WithStack() removes some of the work that would have been done, but doesn't add any work that wasn't already being done.

LGTM

**Pull request merged**